src/patterns/gen-ai/aws-aoss-cw-dashboard/index.ts

```diff
@@ -0,0 +1,374 @@
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+import { Duration, Aws } from 'aws-cdk-lib';
+import {
+  Metric,
+  Dashboard,
+  GraphWidget,
+  GraphWidgetView,
+  Stats,
+  TextWidget,
+} from 'aws-cdk-lib/aws-cloudwatch';
+import { CfnCollection } from 'aws-cdk-lib/aws-opensearchserverless';
+import { Construct } from 'constructs';
+
+
+/**
+ * The properties for the CollectionMonitoringProps class.
+ */
+export interface CollectionMonitoringProps {
+  /*
+    * Optional - The period over which the specified statistic is applied.
+    * @default - 1 hour
+    */
+  readonly period?: Duration;
+
+  /*
+    * Optional - The client ID to use.
+    * @default - Account ID
+    */
+  readonly clientId?: string;
+}
+
+/**
+ * The properties for the CollectionMonitoringProps class.
+ */
+export interface IndexMonitoringProps {
+  /*
+    * Optional - The period over which the specified statistic is applied.
+    * @default - 1 hour
+    */
+  readonly period?: Duration;
+  /*
+    * Optional - The client ID to use.
+    * @default - Account ID
+    */
+  readonly clientId?: string;
+}
+
+/**
+ * The properties for the BedrockCwDashboardProps class.
+ */
+export interface AossCwDashboardProps {
+
+  /**
+   * Optional An existing dashboard where metrics will be added to.
+   * If not provided, the construct will create a new dashboard
+   *
+   * @default - none
+   */
+  readonly existingDashboard?: Dashboard;
+
+  /**
+   * Optional A name for the dashboard which will be created.
+   * If existingDashboard is defined, this value will be ignored.
+   * If not provided, the construct will create a new dashboard named 'BedrockMetricsDashboard'
+   *
+   * @default - none
+   */
+  readonly dashboardName?: string;
+}
+
+/**
+ * The BedrockCwDashboard class.
+ */
+export class AossCwDashboard extends Construct {
+
+  /**
+   * Returns the instance of CloudWatch dashboard used by the construct
+   */
+  public readonly dashboard: Dashboard;
+
+  /**
+   * Constructs a new instance of the AossCwDashboard class.
+   * @param {cdk.App} scope - represents the scope for all the resources.
+   * @param {string} id - this is a a scope-unique id.
+   * @param {AossCwDashboardProps} props - user provided props for the construct.
+   * @since 0.0.0
+   * @public
+   */
+  constructor(scope: Construct, id: string, props: AossCwDashboardProps) {
+    super(scope, id);
+
+    this.dashboard = props.existingDashboard ?? new Dashboard(this, 'AossMetricsDashboard', {
+      dashboardName: props.dashboardName ?? 'AossMetricsDashboard',
```

Comment:
The dashboard name is global. It may give resource name conflict with deployments in different account. suggest to add unique stack id.

Reply:
done in [26e1e8e](https://github.com/awslabs/generative-ai-cdk-constructs/pull/686/commits/26e1e8ef8e017a1305cd8dc5ed9c592bed524004)[26e1e8e](https://github.com/awslabs/generative-ai-cdk-constructs/pull/686/commits/26e1e8ef8e017a1305cd8dc5ed9c592bed524004)

Reply:
Actually reverted it in [5bde878](https://github.com/awslabs/generative-ai-cdk-constructs/pull/686/commits/5bde8789c025e474c85825078bdf9568f3da1600)[5bde878](https://github.com/awslabs/generative-ai-cdk-constructs/pull/686/commits/5bde8789c025e474c85825078bdf9568f3da1600)[5bde878](https://github.com/awslabs/generative-ai-cdk-constructs/pull/686/commits/5bde8789c025e474c85825078bdf9568f3da1600) as stack id provides invalid characters when deploying the stack. Dashboard name can be change through stack props